backend/logic/printful/sync/stockUpdate.js

```diff
@@ -0,0 +1,66 @@
+const _get = require('lodash/get')
+const program = require('commander')
+const { decryptConfig } = require('../../../utils/encryptedConfig')
+
+const { get } = require('../api')
+const { Product } = require('../../../models')
+
+/**
+ * Updates discontinued variants and variants that are out of stock to a collection.
+ * And creates collection if it doesn't exist
+ *
+ * @param {Object} shop
+ * @param {Number|String} productId
+ * @param {Array<Object>} variants
+ */
+const stockUpdate = async (shop, productId, variants) => {
+  const shopConfig = decryptConfig(shop.config)
+  const shopInventory = _get(shopConfig, 'inventory')
+  if (shopInventory) {
+    const variantsStock = variants.reduce(
+      (stockData, variant) => ({
+        ...stockData,
+        [variant]: 0
+      }),
+      {}
+    )
+
+    const apiKey = shopConfig.printful
+    let existingVariants = 0
+
+    if (apiKey) {
+      const productData = await get(`/store/products/${productId}`, {
+        apiKey: apiKey
+      })
+      existingVariants = _get(productData, 'result.sync_product.synced', 0) - variants.length
+      existingVariants = existingVariants < 0 ? 0 : existingVariants
+    }
+
+    const updatedStockData = {
+      productId: productId.toString(),
+      shopId: shop.id,
+      stockLeft: existingVariants * 10000,
```

Comment:
Add a comment as to why we do his math.

Overall I'm still unsure this is the best way to go.

Not being super familiar with the overall Printfu sync flow, my naive thoughts would be that at a high level the logic would be as follow:
  1. Printful calls this webhook with a list of variants that are out of stock or discontinued.
  2. We load the corresponding Product row from the database
  3. We update the field Product.variantsStock to set quantity to zero for all variants  out of stock/discontinued
  4. We update the field Product.stockLeft by recomputing the sum of all variants in Product.variantsStock

Reply:
I now understand Printful Sync does not send inventory data. makes sense then.